src/utils/eth_transaction.cairo

```diff
@@ -201,22 +206,25 @@ namespace EthTransaction {
         }
         finalize_keccak(keccak_ptr_start, keccak_ptr);
 
-        let gas_price_idx = tx_type + 1;
+        let signer_nonce_idx = 1;
+        let (signer_nonce) = Helpers.bytes_to_felt(
+            sub_items[signer_nonce_idx].data_len, sub_items[signer_nonce_idx].data, 0
+        );
         let (gas_price) = Helpers.bytes_to_felt(
-            sub_items[gas_price_idx].data_len, sub_items[gas_price_idx].data, 0
+            sub_items[signer_nonce_idx + 1].data_len, sub_items[signer_nonce_idx + 1].data, 0
```

Comment:
you still need to use the `tx_type` somewhere otherwise it cannot work!

```
let signer_nonce_idx = 1;
let gas_price_idx = tx_type + signer_nonce;
// ...
```